src/cryptonote_core/blockchain.cpp

```diff
@@ -1044,10 +1044,13 @@ bool Blockchain::switch_to_alternative_blockchain(std::list<block_extended_info>
     reorg_notify->notify("%s", std::to_string(split_height).c_str(), "%h", std::to_string(m_db->height()).c_str(),
         "%n", std::to_string(m_db->height() - split_height).c_str(), "%d", std::to_string(discarded_blocks).c_str(), NULL);
 
-  std::shared_ptr<tools::Notify> block_notify = m_block_notify;
-  if (block_notify)
-    for (const auto &bei: alt_chain)
-      block_notify->notify("%s", epee::string_tools::pod_to_hex(get_block_hash(bei.bl)).c_str(), NULL);
+  std::size_t notify_height = split_height;
+  for (const auto& bei: alt_chain)
+  {
+    for (const auto& notifier: m_block_notifiers)
```

Comment:
Consider to swap the loops

Reply:
Any more information? I wrote it this way because the same block should remain in the CPU data cache for each call. However, this should thrash the CPU instruction cache. Are you saying you think the opposite is preferred? Some other reason?

Reply:
In case of empty `m_block_notifiers`, we could skip iteration over `alt_chain`.

Reply:
Done.